src/main/java/de/tum/in/www1/artemis/service/compass/CompassService.java

```diff
@@ -237,72 +257,159 @@ public void addAssessment(long exerciseId, long submissionId, List<Feedback> mod
     }
 
     /**
-     * Get the assessment for a given model from the calculation engine, create an automatic result from it and save it to the database. This is done only if the submission is not
-     * manually assessed already, i.e. the assessment type is not MANUAL and the assessor is not set. Note, that Compass tries to automatically assess every model as much as
+     * Get the automatic result generated by Compass from the automatic result map for the submission with the given id.
+     *
+     * @param submissionId the id of the submission for which to get the automatic result
+     * @return the automatic result for the submission with the given id
+     */
+    public Result getAutomaticResultForSubmission(long submissionId) {
+        return automaticResultMap.get(submissionId);
+    }
+
+    /**
+     * Remove the automatic result generated by Compass from the automatic result map for the submission with the given id.
+     *
+     * @param submissionId the id of the submission for which to remove the automatic result from the map
+     */
+    public void removeAutomaticResultForSubmission(long submissionId) {
+        automaticResultMap.remove(submissionId);
+    }
+
+    /**
+     * Update the (existing) automatic result for the given submission with automatic feedback generated by Compass. If there is no existing result, a new one is created first. The
+     * updated result gets added to the hash map containing all automatic results. Note, that Compass tries to automatically assess every model as much as possible, but does not
+     * submit any automatic assessment to the student. A user has to review every(!) automatic assessment before completing and submitting the assessment manually, even if Compass
+     * could assess 100% of the model automatically.
+     *
+     * @param submissionId the id of the modeling submission for which an automatic result should be generated/updated
+     * @param exerciseId   the id of the corresponding exercise
+     */
+    private void assessAutomatically(long submissionId, long exerciseId) {
```

Comment:
Why is this method even invoked for unsupported modeling exercises? Can we prevent this earlier?

Reply:
I added the `isSupported` check to all relevant public methods of the `CompassService` and removed it from the private methods.